llvm/lib/Target/X86/X86AsmPrinter.cpp

```diff
@@ -181,8 +181,26 @@ void X86AsmPrinter::emitKCFITypeId(const MachineFunction &MF) {
   // Embed the type hash in the X86::MOV32ri instruction to avoid special
   // casing object file parsers.
   EmitKCFITypePadding(MF);
+
+  Register MovReg = X86::EAX;
+  const auto &Triple = MF.getTarget().getTargetTriple();
+  if (Triple.isArch64Bit() && Triple.isOSLinux()) {
```

Comment:
Is this check necessary? I notice that all of the x86-based kCFI tests use the `x86_64-unknown-linux-gnu` triple. If 64-bit Linux is the only supported x86 subtarget then I think this check is redundant. But I'm not 100% certain.

Reply:
You can use assert to guarantee it won't be broken in the future.

Reply:
@phoebewang Good suggestion! I updated the PR to use an assert.